embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/spans/EmbraceExtensions.kt

```diff
@@ -70,10 +71,14 @@ fun SpanEvent.hasFixedAttribute(fixedAttribute: FixedAttribute): Boolean =
 
 fun List<Attribute>.findAttributeValue(key: String): String? = singleOrNull { it.key == key }?.data
 
-fun io.embrace.android.embracesdk.internal.payload.Span.getSessionProperty(key: String): String? {
+fun Span.getSessionProperty(key: String): String? {
     return attributes?.findAttributeValue(key.toSessionPropertyAttributeName())
 }
 
+@Suppress("UNCHECKED_CAST")
+fun Span.getSessionProperties(): Map<String, String> =
+    attributes?.filter { it.key != null && it.data != null }?.associate { it.key to it.data } as Map<String, String>
```

Comment:
I didn't see anywhere this function was used in review & am not sure of its purpose. It seems like either the name or functionality is wrong as the session properties don't live exclusively on a span's attributes?

Reply:
Oh I hope I was able to get rid of this. I needed to cast Map<String?, String?> into Map<String, String> at one point and if I didn't have to, I'd love to get rid of this

Reply:
I think you might've been looking at an older version? I added usage after the first commit - in line 75 of PayloadResurrectionServiceImpl.kt